testdata/p4_16_samples_outputs/invalid-hdr-warnings7.p4-stderr

```diff
@@ -52,9 +52,3 @@ invalid-hdr-warnings7.p4(56): [--Wwarn=invalid_header] warning: accessing a fiel
 invalid-hdr-warnings7.p4(57): [--Wwarn=invalid_header] warning: accessing a field of an invalid header us1[1].h1
                  ^ us1[0].h2.a ^ us1[1].h1.a ^ us1[1].h2.a;
                                  ^^^^^^^^^
-invalid-hdr-warnings7.p4(78): [--Wwarn=invalid_header] warning: accessing a field of an invalid header u.h1
```

Comment:
From the comments in testcase, it looks like line :78 is not expected to be invalid and this change in output is OK.
Infact, as p4_16 spec section https://p4.org/p4-spec/docs/P4-16-working-spec.html#sec-expr-hu,  I would think even line:79 is not invalid as assignment to header union element implicitly sets it to Valid.

```
 78         u.h1.a = 1;
 79         u.h2.a = 1;        // expected invalid

```

However, these warnings are coming from frontend. The changes in this PR does not involve anything which should affect the output from frontend, so I am not sure what change in this PR is causing it to disappear. I will dig further up.

Reply:
I built the compiler from the main branch and ran this test with p4test. The warning for line:78 is not issued.

Perhaps the output  in -stderr is not up-to-date and is not being checked as this is marked as XFAIL in p4test. Since, the FlattenHeaderUnion pass (which introduces conditional statements) is moved after Predication pass, the errors related to conditional statements in actions are not issued.